inference-engine/include/ie_blob.h

```diff
@@ -578,13 +578,14 @@ class TBlob : public MemoryBlob {
      *@brief Virtual destructor.
      */
 #ifdef __clang__
-    virtual ~TBlob();
+    virtual ~TBlob() {};
```

Comment:
That breaks RTTI for Android (see https://android.googlesource.com/platform/ndk/+/master/docs/user/common_problems.md#rtti_exceptions-not-working-across-library-boundaries). So keep implementation in the .cpp file.

Reply:
I think the root cause of this issue is, different type of compiler is used to build the library (IE) and the user program. In this case, IE libraries will be built by MSVC and the user code is built by a Clang based DPC++. I can add something like 'InferenceEngine::TBlob ~TBlob() {};' into an IE CPP source file with __clang__ conditional directive, the code won't be compiled because MSVC will be used for building IE.
Instead, I added conditional directive in ie_blob.h to check if the compiler is DPC++ or not. It requires user code to include <CL/sycl.hpp> so that the __SYCL_COMPILER_VERSION macro is defined.
I have pushed this fix. Please take a look.